crates/l2/proposer/mod.rs

```diff
@@ -108,26 +106,28 @@ impl Proposer {
                     "Failed to get block by hash from storage".to_string(),
                 ))?;
 
-            let withdrawal_txs: Vec<&Transaction> = block
+            let withdrawal_txs: Vec<H256> = block
                 .body
                 .transactions
                 .iter()
-                .filter(|tx| match tx {
-                    Transaction::PrivilegedL2Transaction(tx) => {
-                        tx.tx_type == PrivilegedTxType::Withdrawal
+                .filter_map(|tx| match tx {
+                    Transaction::PrivilegedL2Transaction(priv_tx)
+                        if priv_tx.tx_type == PrivilegedTxType::Withdrawal =>
+                    {
+                        let to = match priv_tx.to {
+                            TxKind::Call(to) => to,
+                            _ => panic!("Withdrawal transaction should be a Call"),
+                        };
+                        let value = &mut [0u8; 32];
+                        priv_tx.value.to_big_endian(value);
+                        Some(keccak(
+                            vec![to.as_bytes(), value, tx.compute_hash().as_bytes()].concat(),
```

Comment:
Nit. this doesn't need to be a `Vec`.
```suggestion
                            [to.as_bytes(), value, tx.compute_hash().as_bytes()].concat(),
```

Reply:
Done [5ec12b6](https://github.com/lambdaclass/lambda_ethereum_rust/pull/961/commits/5ec12b6c6f2c8bd8a6b7ae386b3e324ee4e25a24)